source/extensions/common/proxy_protocol/proxy_protocol_header.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/address.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace ProxyProtocol {
+
+// See https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt for definitions
+
+constexpr char PROXY_PROTO_V1_SIGNATURE[] = "PROXY ";
```

Comment:
These constants overlap with ones created by the proxy proto listener filter. I opted to not refactor the listener filter to use these common ones for now to keep this PR simple. I can do the refactor now if requested however.

Reply:
How about a TODO so we don't lose track?